src/Tree.cpp

```diff
@@ -8,19 +8,20 @@
 #endif
 
 Tree::
-Tree(const Rectangle r, int levels)
+Tree(const Rectangle& r, int levels)
 {
   rectangle_ = r;
-  cells_ = nullptr;
+  // cells_ = nullptr;
+  cells_.empty();
 
   subdivide(levels);
 }
 
 Tree::
 ~Tree()
 {
-  if(cells_ != nullptr) {
-    cells_.reset();
+  if(cells_.size() != 0) {
```

Comment:
The vector will automatically be destroyed, so it is not needed to do anything with `cells_` in the destructor.